js/latoken.js

```diff
@@ -0,0 +1,635 @@
+        return this.fetchOrdersByStatus (symbol, 'active', since, limit, params);
+    }
+
+    async fetchCancelledOrders (symbol = undefined, since = undefined, limit = 50, params = {}) {
+        return this.fetchOrdersByStatus (symbol, 'cancelled', since, limit, params);
+    }
+
+    async fetchFilledOrders (symbol = undefined, since = undefined, limit = 50, params = {}) {
+        return this.fetchOrdersByStatus (symbol, 'filled', since, limit, params);
+    }
+
+    async fetchPartiallyFilledOrders (symbol = undefined, since = undefined, limit = 50, params = {}) {
+        return this.fetchOrdersByStatus (symbol, 'partiallyFilled', since, limit, params);
+    }
+
+    async fetchOrdersByStatus (symbol, status, since = undefined, limit = 100, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        let request = {
+            'status': status,
+        };
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['symbol'];
+        }
+        if (limit !== undefined)
+            request['limit'] = limit;
+        let response = await this.privateGetOrderStatus (this.extend (request, params));
+        let orders = this.parseOrders (response);
+        return orders;
+    }
+
+    async fetchAllActiveOrders (symbol, limit = 50, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'symbol': symbol,
+            'limit': limit,
+        };
+        request['timestamp'] = this.nonce ();
+        let response = await this.privateGetOrderActive (this.extend (request, params));
+        let orders = this.parseOrders (response);
+        return orders;
+    }
+
+    async fetchOrder (id, params = {}) {
+        await this.loadMarkets ();
+        let request = {
+            'orderId': id,
+        };
+        let response = await this.privateGetOrderGetOrder (this.extend (request, params));
+        return (this.parseOrder (response));
+    }
+
+    parseNewOrder (response) {
+        return ({
+            'orderId': response['orderId'],
+            'cliOrdId': response['cliOrdId'],
+            'pairId': this.safeFloat (response, 'pairId'),
+            'symbol': response['symbol'],
+            'side': response['side'],
+            'orderType': response['orderType'],
+            'price': this.safeFloat (response, 'price'),
+            'amount': this.safeFloat (response, 'amount'),
+        });
+    }
+
+    async createOrder (symbol, cliOrdId, side, price, amount, orderType, timeAlive, params = {}) {
+        let order = {
+            'symbol': symbol,
+            'cliOrdId': cliOrdId,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'orderType': orderType,
+            'timeAlive': timeAlive,
+        };
+        let result = await this.privatePostOrderNew (this.extend (order, params));
+        return this.parseNewOrder (result);
```

Comment:
This should `return this.parseOrder (result)` like we do in other exchanges, and `parseNewOrder` should be unified with `parseOrder` (in fact for parsing all orders we need just the `parseOrder`, all other order parsing methods should be merged into it – because it's the same order structure with some fields missing occasionally, and this is perfectly fine with safe-methods. No need to duplicate the code for parsing the same fields all over the file.

The rule of thumb is:

- methods returning a single order should `return this.parseOrder (order, market);`, this is true for `createOrder`, `cancelOrder`
- methods returning many orders should `return this.parseOrders (orders, market, since, limit);`, this is true for `fetchOrders`, `fetchClosedOrders`, `fetchOpenOrders` – in these cases we should use the base class `this.parseOrders (...)` and we should not redefine it in the derived class. The base `parseOrders` method will call `parseOrder` for each element of the incoming array.

↑ The same logic is true for trades ↓

- `parseTrade` defined in the derived class (just as `parseOrder`)
- `parseTrades` from the base class (just as `parseOrders`)

Reply:
done